analyses/org.eclipse.tracecompass.incubator.executioncomparision.core/src/org/eclipse/tracecompass/incubator/internal/executioncomparision/core/DifferentialSeqCallGraphAnalysis.java

```diff
@@ -0,0 +1,377 @@
+    }
+
+    /**
+     * Refresh differential call graph
+     *
+     * @param monitor
+     *            the progress monitor
+     */
+    public void refreshDiffCG(@Nullable IProgressMonitor monitor) {
+        try (ScopeLog sl = new ScopeLog(LOGGER, Level.CONFIG, "DifferentialSequenceCGA::refresh()")) { //$NON-NLS-1$
+            Collection<WeightedTree<ICallStackSymbol>> originalTree = new ArrayList<>();
+            Collection<WeightedTree<ICallStackSymbol>> diffTree = new ArrayList<>();
+            /*
+             * if (fEndA == null || fStartA == null) { fEndA =
+             * TmfTimestamp.BIG_CRUNCH; } if (fStartA == null) { fStartA =
+             * TmfTimestamp.BIG_BANG; } if (fEndB == null) { fEndB =
+             * TmfTimestamp.BIG_CRUNCH; } if (fStartB == null) { fStartB =
+             * TmfTimestamp.BIG_BANG; }
+             */
+            WeightedTreeSet<ICallStackSymbol, Object> callGraphA = mergeCallGraph(fStartA, fEndA, fTraceListA);
+            Collection<@NonNull ?> processes = callGraphA.getTreesForNamed(MERGE);
+            for (Object process : processes) {
+                originalTree.add((AggregatedCalledFunction) process);
+            }
+
+            WeightedTreeSet<ICallStackSymbol, Object> callGraphB = mergeCallGraph(fStartB, fEndB, fTraceListB);
+            processes = callGraphB.getTreesForNamed(MERGE);
+            for (Object process : processes) {
+                diffTree.add((AggregatedCalledFunction) process);
+            }
+
+            Collection<DifferentialWeightedTree<ICallStackSymbol>> trees;
+            if (diffTree.isEmpty()) {
+                trees = ParametricWeightedTreeUtils.diffTrees(diffTree, originalTree, fStatistic);
+            } else {
+                trees = ParametricWeightedTreeUtils.diffTrees(originalTree, diffTree, fStatistic);
+            }
+
+            IWeightedTreeProvider<ICallStackSymbol, ICallStackElement, AggregatedCallSite> instrumentedCallStackAnalysis = Iterables.get(fTraceCallGraphRegistry.values(), 0);
+            setDifferentialCallGraphProvider(new DifferentialCallGraphProvider(instrumentedCallStackAnalysis, trees));
+        }
+    }
+
+    /**
+     * Merge callgraph
+     *
+     * @param start
+     *            the start time
+     * @param end
+     *            the end time stamp
+     * @param traceList
+     *            the list of traces to merge
+     * @return WeightedTreeSet<ICallStackSymbol, Object> the merged call graph
+     *
+     */
+    public WeightedTreeSet<ICallStackSymbol, Object> mergeCallGraph(ITmfTimestamp start, ITmfTimestamp end, List<String> traceList) {
+        try (ScopeLog sl = new ScopeLog(LOGGER, Level.FINE, "DifferenentialSequenceCGA::MergeCallGraph")) { //$NON-NLS-1$
+            List<CallGraph> cGList = new ArrayList<>();
+            WeightedTreeSet<ICallStackSymbol, Object> newTreeSet = new WeightedTreeSet<>();
+            String mainGroup = MERGE;
+
+            for (String traceName : traceList) {
```

Comment:
this can't be null. In initialization step (    private List<String> fTraceListA = new ArrayList<>(); private List<String> fTraceListB = new ArrayList<>();) traceLists are initialized as a empty list. And they will be filled by the signal handling before calling this function. However if it is null, the function returns an empty weightedTreeSet.